docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -19,32 +15,167 @@ We'll do our best to keep things easy to follow, and as sequential as possible,
 This documentation page covers the _how_ of migrating from v4 to v5.
 The _why_ will be covered in an upcoming blog post on Medium.
 
-## Updating your dependencies
+## Migration Steps
+
+- [Update React & TypeScript](#update-react-&-typescript-version)
+- [ThemeProvider Setup](#themeprovider-setup)
+- [Update Material-UI](#update-material-ui-version)
+- [Run Codemod](#run-codemod)
+  - [🪄 preset-safe](#preset-safe)
+  - [variant-prop (optional)](#variant-prop)
+  - [link-underline-hover (optional)](#link-underline-hover)
+- [Handling Breaking Changes](#handling-breaking-changes)
+- [Migrate `makeStyles` to emotion](#migrate-makestyles-to-emotion)
+- [Troubleshooting](#troubleshooting)
+
+> 🛎 always create small commit on any changes to help the migration goes smoother.
+> Please check [Troubleshooting](#troubleshooting) section, if you encounter any issue or [create an issue](https://github.com/mui-org/material-ui/issues/new?assignees=&labels=status%3A+needs+triage&template=1.bug.md) with this title format `[Migration] Summary of your issue`.
+
+## **Update React & TypeScript version**
+
+- The minimum supported version of **React** was increased from v16.8.0 to v17.0.0.
+- The minimum supported version of **TypeScript** was increased from v3.2 to v3.5.
+
+  > We try to align with types released from [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) (i.e. packages published on npm under the `@types` namespace).
+  > We will not change the minimum supported version in a major version of Material-UI.
+  > However, we generally recommend to not use a TypeScript version older than the [lowest supported version of DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped#older-versions-of-typescript-33-and-earlier)
 
-The very first thing you will need to do is to update your dependencies.
+**Note:** if your project includes these packages, please upgrade them to the `latest` version.
 
-### Update Material-UI version
+- `react-scripts`
+- `@types/react`
+- `@types/react-dom`
 
-You need to update your `package.json` to use the latest version of Material-UI and its peer dependencies.
+> ✅ Please make sure that your application is still **running** without error and **commit** the change before continuing the next step.
 
-```json
-"dependencies": {
-  "@emotion/react": "^11.0.0",
-  "@emotion/styled": "^11.0.0",
-  "@material-ui/core": "^5.0.0"
+## **`ThemeProvider` Setup**
+
+Make sure that `MuiThemeProvider` is defined at the root of your application (even if you are using the **default theme**) and **NO** `useStyles` is called before `<MuiThemeProvider>`.
```

Comment:
I thought we renamed it:

```suggestion
Make sure that `ThemeProvider` is defined at the root of your application (even if you are using the **default theme**) and **NO** `useStyles` is called before `<ThemeProvider>`.
```

Reply:
I would also explain why, to not confuse developers. It's for the ones that will keep using `@material-ui/styles`. So maybe the import should be from `@material-ui/styles`, not `@material-ui/core/styles`.

Reply:
This step is still on v4 so I think `MuiThemeProvider` is required.

Reply:
+1 for the name. I would leave the import from `@material-ui/core/styles` as it contains also the emotion/styled-components' theme context. Otherwise, for the core components, they would need to anyway change it later.

Reply:
> This step is still on v4 so I think MuiThemeProvider is required.

Importing `ThemeProvider` from `@material-ui/core/styles` should work in v4 too.

Reply:
updated to `ThemeProvider`